torchft/manager_integ_test.py

```diff
@@ -423,3 +423,46 @@ def test_local_sgd_recovery(self) -> None:
             )
 
         self.assertEqual(failure_injectors[1].count, 1)
+
+    def test_quorum_timeout(self) -> None:
+        with ExitStack() as stack:
+            lighthouse = Lighthouse(
+                bind="[::]:0",
+                min_replicas=2,
+            )
+            stack.callback(lighthouse.shutdown)
+
+            store = dist.TCPStore(
+                host_name="localhost",
+                port=0,
+                is_master=True,
+                wait_for_workers=False,
+            )
+
+            pg = ProcessGroupGloo()
+            manager = Manager(
+                pg=pg,
+                min_replica_size=2,
+                load_state_dict=lambda x: None,
+                state_dict=lambda: None,
+                store_addr="localhost",
+                store_port=store.port,
+                rank=0,
+                world_size=2,
+                lighthouse_addr=lighthouse.address(),
+                port=19530,
+                use_async_quorum=False,
+            )
+            stack.callback(manager.shutdown)
+
+            with self.assertRaisesRegex(
+                TimeoutError,
+                "status: Cancelled, message.*Timeout expired",
+            ):
+                manager.start_quorum(timeout=timedelta(seconds=0.01))
+
+            with self.assertRaisesRegex(
+                TimeoutError,
+                "status: Cancelled, message.*Timeout expired",
+            ):
+                manager.should_commit(timeout=timedelta(seconds=0.01))
```

Comment:
This doesnt really test that methods now have method specific timeouts though. They could still timeout on the default timeout and pass. Maybe set a different timeout in Manager init from the one used by method and assert the method one was used?

Reply:
added an elapsed time check -- the default is 60s so easy to verify that we're shorter